src/coreclr/build-runtime.cmd

```diff
@@ -453,7 +453,7 @@ if %__BuildCrossArchNative% EQU 1 (
         set __ExtraCmakeArgs="-DCMAKE_BUILD_TYPE=!__BuildType!" 
     )
 
-    set __ExtraCmakeArgs=!__ExtraCmakeArgs! %__CMakeClrBuildSubsetArgs% "-DCLR_CROSS_COMPONENTS_BUILD=1" "-DCLR_CMAKE_TARGET_ARCH=%__BuildArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=0" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=0" "-DCLR_ENG_NATIVE_DIR=%__RepoRootDir%/eng/native" "-DCLR_REPO_ROOT_DIR=%__RepoRootDir%" %__CMakeArgs%
+    set __ExtraCmakeArgs=!__ExtraCmakeArgs! %__CMakeClrBuildSubsetArgs% "-DCLR_CROSS_COMPONENTS_BUILD=1" "-DCLR_CMAKE_TARGET_ARCH=%__BuildArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=0" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=0" "-DCLR_ENG_NATIVE_DIR=%__RepoRootDir%/eng/native" "-DCLR_SRC_NATIVE_DIR=%__RepoRootDir%\src\native" "-DCLR_REPO_ROOT_DIR=%__RepoRootDir%" %__CMakeArgs%
```

Comment:
Is it still a good idea to pass all these paths via commend line? I believe that this was a temporary measure as we have been merging the repos. I would expect that these paths can be just defined in the cmake files now.

Reply:
cc @jkoritzinsky

Reply:
I like the idea of putting these paths directly into CMake instead of in the scripts, but IIRC @trylek or @janvorli preferred this model.

Reply:
Agreed, there is no need for `CLR_ENG_NATIVE_DIR` and `CLR_SRC_NATIVE_DIR` to be passed from scripts, we can use `CLR_REPO_ROOT_DIR` to define them in the base cmake file (which is included everywhere). I will take a look.

Reply:
The `mostly-defined-in-cmake-scripts` version looks like this https://github.com/am11/runtime/commit/59c2287a151e884f7fb71d6d29b21420594ab109. I can push that commit on PR branch if it is the consensus. Please let me know. :)

Reply:
I'd be fine with removing those cmake arguments if we defined these variables in a common script included by all the three parts (coreclr, libraries, installer).  We could do the same for the CLR_REPO_ROOT_DIR. 
@am11 looking at your experimental commit, I'd like to keep the CLR_ENG_NATIVE_DIR / CLR_SRC_NATIVE_DIR (in the scripts, not as arguments). Maybe we could create `eng/native/configurepaths.cmake` that would set these variables.

Reply:
>  I'd like to keep the CLR_ENG_NATIVE_DIR / CLR_SRC_NATIVE_DIR (in the scripts, not as arguments)

@janvorli, my thinking was; `CLR_REPO_ROOT_DIR` needs to be passed from the shell script to cmake, because from any cmake file under `src/` that needs to include a file under `eng/native/`, it requires reference to repo root. Otherwise, we will end up including with relative paths (`../../../eng/native` style), which makes the refactoring of code harder.

Reply:
You are right, I have not realized that to find the `configurepaths.cmake` script I was suggesting before already requires knowledge of the root dir location. However, since there are only three places where this file would be included, it would be not be that bad to use relative paths in those places - if we needed to move the coreclr / installer / libraries later, it would require modifying single location.
To be specific, e.g. src/coreclr/CMakeLists.txt would contain:
```cmake
include(../../eng/native/configurepaths.cmake)
```
and the `configurepaths.cmake` would contain
```cmake
file(REAL_PATH CLR_REPO_ROOT_DIR ../.. BASE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
set(CLR_ENG_NATIVE_DIR ${CMAKE_CURRENT_LIST_DIR})
file(REAL_PATH CLR_SRC_NATIVE_DIR ../../src/native BASE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
```
I guess we will need to use the `file(TO_NATIVE_PATH` on those, as you've found in your experiment, unless the REAL_PATH already does that. I hope I got the `file(REAL_PATH` syntax right, I have not used it before.

Reply:
This sketch makes sense. Unfortunately, `REAL_PATH` is _very_ new; introduced in cmake 3.19 (curret latest release), and we want to support old versions like 3.8. So I have used `get_filename_component(VAR  PATH ABSOLUTE)` instead.

Reply:
Ah, I haven't realized that REAL_PATH is new. Great that you have found an alternative